### PR TITLE
big bux fixing

### PR DESCRIPTION
**Pull request by 77seven77seven**
# more stuff to come

## About the Pull Request

i was bored so i decided to fix like 3 issues

## Why It's Good For The Game

i hate bugs

## Changelog

:cl:
add: Cameras now use balloon alerts instead of to_chats
code: camera code is now a little more cleaner holy shit
add: Credits now scroll up properly, and have less immersion breaking text.
admin: The player panel now shows all mobs with a ckey instead of just humans
del: The medical vending machine in the containment chamber of SCP-049 no longer has an access requirement.
imageadd: The telebaton now has inhand sprites.
/:cl:

### CODE REVIEW

File: tools/CreditsTool/CreditsTool.sln [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (Tsurupeta) [inline tools/CreditsTool/CreditsTool.sln]: wtf is this?

File: code/game/machinery/camera/camera.dm [hunk 2]
```diff
@@ -212,12 +211,14 @@
...
 		var/itemname = X.name
 		var/info = X.info
-		to_chat(U, "You hold \a [itemname] up to the camera ...")
+		user.balloon_alert(U, "You hold \a [itemname] up to the camera...")
 		for(var/mob/living/silicon/ai/O in GLOB.living_mob_list_)
-			if(!O.client) continue
-			if(U.name == "Unknown") to_chat(O, "<b>[U]</b> holds \a [itemname] up to one of your cameras ...")
-			else to_chat(O, "<b><a href='byond://?src=\ref[O];track2=\ref[O];track=\ref[U];trackname=[U.name]'>[U]</a></b> holds \a [itemname] up to one of your cameras ...")
-			show_browser(O, text("<HTML><HEAD><TITLE>[]</TITLE></HEAD><BODY><TT>[]</TT></BODY></HTML>", itemname, info), text("window=[]", itemname))
+			if(!O.client)
```
Comment (Tsurupeta) [inline code/game/machinery/camera/camera.dm:216]: Bruh, you broke the thing

File: code/game/machinery/camera/camera.dm [hunk 3]
```diff
@@ -245,24 +246,23 @@
 	set_status(!src.status)
 	if (!(src.status))
 		if(user)
-			visible_message(SPAN_NOTICE(" [user] has deactivated [src]!"))
+			balloon_alert_to_viewers("[user] has deactivated [src]!")
```
Comment (Tsurupeta) [inline code/game/machinery/camera/camera.dm]: wrong usage of balloon alerts

File: tools/CreditsTool/CreditsTool.sln (hunk 1 above)
Comment (77seven77seven) [reply tools/CreditsTool/CreditsTool.sln]: i have no idea it just showed up

File: code/game/machinery/camera/camera.dm (hunk 3 above)
Comment (77seven77seven) [reply code/game/machinery/camera/camera.dm]: whats the correct use

File: code/game/machinery/camera/camera.dm (hunk 2 above)
Comment (77seven77seven) [reply code/game/machinery/camera/camera.dm:216]: how

File: tools/CreditsTool/CreditsTool.sln (hunk 1 above)
Comment (TheDarkElites) [reply tools/CreditsTool/CreditsTool.sln]: You used visual studio not visual studio code didn't you

Comment (77seven77seven) [reply tools/CreditsTool/CreditsTool.sln]: > You used visual studio not visual studio code didn't you

image

File: code/game/machinery/camera/camera.dm (hunk 2 above)
Comment (Tsurupeta) [reply code/game/machinery/camera/camera.dm:216]: You removed continue

File: code/game/machinery/camera/camera.dm (hunk 3 above)
Comment (Tsurupeta) [reply code/game/machinery/camera/camera.dm]: Not for messages X did Y

Comment (aidan-bowman) [reply code/game/machinery/camera/camera.dm]: [Read TGstation's guide on how to use balloon alerts.](https://hackmd.io/@tgstation/SJIle5wYd#balloon_alertballoon_alert_to_viewers)

File: code/game/machinery/camera/camera.dm [hunk 4]
```diff
@@ -212,12 +211,14 @@
 		var/obj/item/paper/X = W
 		var/itemname = X.name
 		var/info = X.info
-		to_chat(U, "You hold \a [itemname] up to the camera ...")
+		user.balloon_alert(U, "You hold \a [itemname] up to the camera...")
```
Comment (aidan-bowman) [inline code/game/machinery/camera/camera.dm]:
```suggestion
		user.balloon_alert(U, "item held up to camera")
```

File: code/game/machinery/camera/camera.dm (hunk 3 above)
Comment (aidan-bowman) [reply code/game/machinery/camera/camera.dm]:
```suggestion
			balloon_alert_to_viewers("camera deactivated")
```

File: code/game/machinery/camera/camera.dm [hunk 5]
```diff
@@ -245,24 +246,23 @@
...
 		else
-			visible_message(SPAN_NOTICE(" [src] clicks and shuts down. "))
+			balloon_alert_to_viewers("[src] clicks and shuts down.")
 		playsound(src.loc, 'sounds/items/Wirecutter.ogg', 100, 1)
 		icon_state = "[initial(icon_state)]1"
 		add_hiddenprint(user)
 	else
 		if(user)
-			visible_message(SPAN_NOTICE(" [user] has reactivated [src]!"))
+			balloon_alert_to_viewers("[user] has reactivated [src]!")
```
Comment (aidan-bowman) [inline code/game/machinery/camera/camera.dm]:
```suggestion
			balloon_alert_to_viewers("camera reactived")
```

File: code/game/machinery/camera/camera.dm [hunk 6]
```diff
@@ -245,24 +246,23 @@
 	set_status(!src.status)
 	if (!(src.status))
 		if(user)
-			visible_message(SPAN_NOTICE(" [user] has deactivated [src]!"))
+			balloon_alert_to_viewers("[user] has deactivated [src]!")
 		else
-			visible_message(SPAN_NOTICE(" [src] clicks and shuts down. "))
+			balloon_alert_to_viewers("[src] clicks and shuts down.")
```
Comment (aidan-bowman) [inline code/game/machinery/camera/camera.dm]:
```suggestion
			balloon_alert_to_viewers("camera self-deactivated")
```

File: code/game/machinery/camera/camera.dm [hunk 7]
```diff
@@ -245,24 +246,23 @@
...
 		playsound(src.loc, 'sounds/items/Wirecutter.ogg', 100, 1)
 		icon_state = "[initial(icon_state)]1"
 		add_hiddenprint(user)
 	else
 		if(user)
-			visible_message(SPAN_NOTICE(" [user] has reactivated [src]!"))
+			balloon_alert_to_viewers("[user] has reactivated [src]!")
 		else
-			visible_message(SPAN_NOTICE(" [src] clicks and reactivates itself. "))
+			balloon_alert_to_viewers("[src] clicks and reactivates itself.")
```
Comment (aidan-bowman) [inline code/game/machinery/camera/camera.dm]:
```suggestion
			balloon_alert_to_viewers("camera self-reactivated")
```

File: code/game/machinery/camera/camera.dm [hunk 8]
```diff
@@ -381,7 +381,7 @@
 		return 0
 
 	if(WT.remove_fuel(0, user))
-		to_chat(user, SPAN_NOTICE("You start to weld \the [src].."))
+		user.balloon_alert(user, "You start to weld \the [src]...")
```
Comment (aidan-bowman) [inline code/game/machinery/camera/camera.dm]:
```suggestion
		user.balloon_alert(user, "welding...")
```

File: code/game/machinery/camera/camera.dm [hunk 9]
```diff
@@ -245,24 +246,23 @@
...
-			visible_message(SPAN_NOTICE(" [src] clicks and reactivates itself. "))
+			balloon_alert_to_viewers("[src] clicks and reactivates itself.")
 		playsound(src.loc, 'sounds/items/Wirecutter.ogg', 100, 1)
 		icon_state = initial(icon_state)
 		add_hiddenprint(user)
 
 /obj/machinery/camera/proc/take_damage(force, message)
-	//prob(25) gives an average of 3-4 hits
-	if (force >= toughness && (force > toughness*4 || prob(25)))
+	if (force >= toughness && (force > toughness*4))
```
Comment (aidan-bowman) [inline code/game/machinery/camera/camera.dm:266]: Did you actually read what you wrote here?

File: code/game/machinery/camera/camera.dm [hunk 10]
```diff
@@ -196,11 +195,11 @@
 				assembly.dir = src.dir
 				if(stat & BROKEN)
 					assembly.state = 2
-					to_chat(user, SPAN_NOTICE("You repaired \the [src] frame."))
+					user.balloon_alert(user, "You repaired \the [src] frame.")
 					cancelCameraAlarm()
 				else
 					assembly.state = 1
-					to_chat(user, SPAN_NOTICE("You cut \the [src] free from the wall."))
+					user.balloon_alert(user, "You cut \the [src] free from the wall.")
```
Comment (aidan-bowman) [inline code/game/machinery/camera/camera.dm]:
```suggestion
					user.balloon_alert(user, "camera removed from wall")
```

File: code/game/machinery/camera/camera.dm [hunk 11]
```diff
@@ -196,11 +195,11 @@
 				assembly.dir = src.dir
 				if(stat & BROKEN)
 					assembly.state = 2
-					to_chat(user, SPAN_NOTICE("You repaired \the [src] frame."))
+					user.balloon_alert(user, "You repaired \the [src] frame.")
```
Comment (aidan-bowman) [inline code/game/machinery/camera/camera.dm]:
```suggestion
					user.balloon_alert(user, "camera repaired")
```

File: code/game/machinery/camera/camera.dm [hunk 12]
```diff
@@ -178,8 +178,7 @@
 //		to_chat(user, SPAN_NOTICE("You start to [panel_open ? "close" : "open"] the camera's panel."))
 		//if(toggle_panel(user)) // No delay because no one likes screwdrivers trying to be hip and have a duration cooldown
 		panel_open = !panel_open
-		user.visible_message(SPAN_WARNING("[user] screws the camera's panel [panel_open ? "open" : "closed"]!"),
-		SPAN_NOTICE("You screw the camera's panel [panel_open ? "open" : "closed"]."))
+		user.balloon_alert_to_viewers("[user] screws the camera's panel [panel_open ? "open" : "closed"]!", "You screw the camera's panel [panel_open ? "open" : "closed"].")
```
Comment (aidan-bowman) [inline code/game/machinery/camera/camera.dm]:
```suggestion
		user.balloon_alert_to_viewers("camera panel screwed [panel_open ? "open" : "closed"]")
```

File: code/game/movietitles.dm [hunk 13]
```diff
@@ -185,23 +185,23 @@ GLOBAL_LIST(end_titles)
 
 	titles += "<center>[jointext(staff,"<br>")]</center>"
 	if(goodboys.len)
-		titles += "<center>STAFF'S GOOD BOYS:<br>[english_list(goodboys)]</center><br>"
+		titles += "<center>STAFF'S GOOD BOYS:<br>[english_list(goodboys)]</center><br>" //wtf are goodboys? what is this fetish shit
```
Comment (aidan-bowman) [inline code/game/movietitles.dm]:
```suggestion
		titles += "<center>STAFF'S GOOD BOYS:<br>[english_list(goodboys)]</center><br>"]
```

don't do this PLEASE

**Review by aidan-bowman — CHANGES_REQUESTED** [review]

File: code/game/machinery/camera/camera.dm (hunk 9 above)
Comment (77seven77seven) [reply code/game/machinery/camera/camera.dm:266]: lol no

File: tools/CreditsTool/CreditsTool.sln (hunk 1 above)
Comment (77seven77seven) [reply tools/CreditsTool/CreditsTool.sln]: removed it

File: code/game/machinery/camera/camera.dm (hunk 2 above)
Comment (Tsurupeta) [reply code/game/machinery/camera/camera.dm:216]: Check indentation

**Pull request closed without merging**